task04/storage.h

```diff
@@ -1,16 +1,49 @@
-
+#include <cstring>
 
 template <typename T>
 class Storage
 {
 public:
     Storage() = default;
-    Storage(const T &val);
-    ~Storage();
+    Storage(const T &val): m_val(val) {};
+    ~Storage(){}
 
-    void SetVal(const T &val);
-    T GetVal();
+    void SetVal(const T &val)
+    {
+    	m_val = val;
+    }
+    T GetVal()
+    {
+    	return m_val;
+    }
 private:
     T m_val;
 };
 
+template <>
+class Storage<char *>
+{
+public:
+    Storage() = default;
+    Storage(const char* val)
+    {
+    	m_val = (char*)malloc(sizeof(val));
```

Comment:
sizeof() always returns 8 here as size of pointer. 
Should be used strlen() instead.